cmd/domain-mapping-webhook/main.go

```diff
@@ -0,0 +1,86 @@
+package main
```

Comment:
I wonder if we can fold controller's and webhooks into one pod with the leader-election work being done :thinking:. It'd conflate scaling of one with the other, but maybe that's fine for extensions like this?

cc @mattmoor

Reply:
is there a big advantage to doing that vs. keeping them separate? As you say it seems a shame to lose the ability to scale them independently (ideally the webhook doesn't need to have a cache of every ingress in the system for example, and the failure cases -> probes/PDBs are pretty different), plus when this lands they'll be separate so seems easier to keep them that way.

Reply:
The short answer is yes, and it's how I've run mink ~forever for reasons we don't need to dive into here.

The longer answer is that I see this as a downstream distribution problem and I'd like to enable either or, and my bias is generally towards the most flexible (unopinionated) upstream option.

Reply:
Totally see the point in mink fwiw where the goal is to have one simple thing to deploy, for _this_ it seems a little more flexible to have them separate, especially since that's how this will eventually land anyway when we lose the alpha flag.

Reply:
Ack, didn't want to block anything.